server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -170,6 +176,79 @@ public List<ScimUser> retrieveByUsernameAndZone(String username, String zoneId)
         return jdbcTemplate.query(USER_BY_USERNAME_AND_ZONE_QUERY , mapper, username, zoneId);
     }
 
+    /**
+     * This method should behave similar to {@link JdbcScimUserProvisioning#query(String, String, boolean, String)},
+     * with the filtering for active IdPs (if required) being the only difference.
+     */
+    @Override
+    public List<ScimUser> retrieveByScimFilter(
```

Comment:
Is there a refactor you can do to reduce the number of arguments here to 3 ro 4? For example, in the implementation, the path for includeInactive true and false and completely independent, so it would make sense to make those two different methods.

Reply:
Fixed with commit 709a372